cdap-watchdog/src/main/java/co/cask/cdap/logging/serialize/LoggingEventSerializer.java

```diff
@@ -107,6 +114,16 @@ public long decodeEventTimestamp(ByteBuffer buffer) throws IOException {
     throw new IOException("Missing timestamp field in the LoggingEvent schema");
   }
 
+  private BinaryDecoder getDecoder(ByteBuffer buffer) {
+    if (buffer.hasArray()) {
+      decoder = DecoderFactory.get().binaryDecoder(buffer.array(), buffer.arrayOffset() + buffer.position(),
+                                                   buffer.remaining(), decoder);
+    } else {
+      decoder = DecoderFactory.get().binaryDecoder(Bytes.toBytes(buffer), decoder);
+    }
+    return decoder;
```

Comment:
`decodeEventTimestamp` above has duplicate code , it can instead call `decoder = getDecoder(buffer)` now.